packages/cli/src/cmds/agentInstaller/agentProcedure.ts

```diff
@@ -108,6 +106,151 @@ export default abstract class AgentProcedure {
     return validationResult;
   }
 
+  async filesLocallyModified(): Promise<string[]> {
+    let files: string[] = [];
+
+    let gitStatus;
+    try {
+      const stdout = runSync(new CommandStruct('git', ['status', '-s'], this.installer.path));
+      if (stdout.length > 0) {
+        const lines = stdout.split('\n');
+        for (const line of lines) {
+          gitStatus = line.substring(0, 2);
+          const file = line.substring(3);
+          if (gitStatus[0] !== '?') {
+            files.push(file);
+          }
+        }
+      } else {
+        gitStatus = '[no status]';
+      }
+    } catch (e) {
+      const gitError = (e as Error).message.split('\n')[1];
+      gitStatus = `[git status failed, ${gitError}]`;
+    }
+
+    return files;
+  }
+
+  async gitStatus(file: string): Promise<string> {
+    let gitStatus;
+    try {
+      const stdout = runSync(new CommandStruct('git', ['status', '-s', file], this.installer.path));
+      if (stdout.length > 0) {
+        // git status output looks like this:
+        //  M Gemfile
+        //  M Gemfile.lock
+        // ?? appmap.yml
+        gitStatus = stdout.split('\n')[0].substring(0, 2);
+      } else {
+        gitStatus = '[no status]';
+      }
+    } catch (e) {
+      const gitError = (e as Error).message.split('\n')[1];
+      gitStatus = `[git status failed, ${gitError}]`;
+    }
+
+    return gitStatus;
+  }
+
+  async gitCommit(files: string[], commitMessage: string): Promise<boolean> {
+    let gitCommit;
+    let isPartOfProject: boolean = false;
+    let itWorked: boolean = false;
+    try {
+      let params: string[] = ['commit', '-m', commitMessage];
+      for (const file of files) {
+        params.push(file);
+      }
+      const stdout = runSync(new CommandStruct('git', params, this.installer.path));
+      itWorked = true;
+    } catch (e) {
+      const gitError = (e as Error).message.split('\n')[1];
+      gitCommit = `[git commit failed, ${gitError}]`;
+    }
+
+    return itWorked;
```

Comment:
If you `return true` above then you can either `return false` here, or let it implicitly return `undefined` which will be falsey.